cmd/push.go

```diff
@@ -155,46 +153,42 @@ func runPush(ctx context.Context, dev *model.Dev, autoDeploy bool, imageTag, okt
 		}
 	}
 
-	trList, err := apps.GetTranslations(ctx, dev, k8sObject, false, c)
+	trList, err := apps.GetTranslations(ctx, dev, app, false, c)
 	if err != nil {
 		return err
 	}
 
 	for _, tr := range trList {
-		if tr.K8sObject == nil {
-			continue
-		}
-
 		if len(dev.Services) == 0 {
-			if tr.K8sObject.GetAnnotation(model.OktetoAutoCreateAnnotation) == model.OktetoUpCmd || tr.K8sObject.PodTemplateSpec.Spec.Containers[0].Name == "dev" {
-				tr.K8sObject.SetAnnotation(model.OktetoAutoCreateAnnotation, model.OktetoPushCmd)
+			if tr.App.Annotations()[model.OktetoAutoCreateAnnotation] == model.OktetoUpCmd || tr.App.PodSpec().Containers[0].Name == "dev" {
+				tr.App.Annotations()[model.OktetoAutoCreateAnnotation] = model.OktetoPushCmd
```

Comment:
Minor thing: This syntax to set the value of an annotation could be a bit confusing. Probably is for coming from Java, but if I see `App.Annotations()` I would expect to have a local copy of the annotations and adding a new annotation wouldn't modify the original annotations map.

I see more intuitive something like:

```
// Convert function into a property
tr.App.Annotations[model.OktetoAutoCreateAnnotation] = model.OktetoPushCmd
``` 

or 

```
tr.App.SetAnnotation(model.OktetoAutoCreateAnnotation, model.OktetoPushCmd)
```

Anyways, just a minor comment, not a strong opinion, you can leave it as it is

Reply:
Great point!
I want to do another pass to the code and do final testing, so I will go with the `SetAnnotation` and `GetAnnotation` approach, I agree it is clearer